### ISSUE TITLE
Attempting to tab complete after '${ errors

### ISSUE THREAD

**Issue by neerolyte** [scop/bash-completion#189]
Something in bash-completion appears to break after trying to tab complete `'${` in a file position e.g

```
$ ls -- '${-bash: unexpected EOF while looking for matching `}'
-bash: syntax error: unexpected end of file
```

```
$ rm -- '${-bash: unexpected EOF while looking for matching `}'
-bash: syntax error: unexpected end of file
```

I had a directory in my `$PWD` at the time named like `${FOO}` but this isn't required for the unexpected EOF error to show up.

**Comment by OliverLew** [comment]
It might be the same issue, here a case that resulted in similar error message for the first time, crashed the second time:

```
$ "$(dirname "$(base[Tab the first time]bash: unexpected EOF while looking for matching `)'
$ "$(dirname "$(base[Tab the second time]free(): double free detected in tcache 2
Aborted (core dumped)
```

Actually, how do I *quickly* test bash without bash-completion? On arch linux I uninstalled the `bash-completion` package, but the bug still appears. I tried fresh start from a new terminal or tty, and reboot, and `bash --noprofile --norc`.

Since my case only involves in command completion, could this be a bash bug?

**Comment by akinomyoga** [comment]
> Does this mean that my case is not the same?

@OliverLew I think this is a different issue. The reasons are

1. First, as you have reported in `bug-bash` mailing list ([2021-03/msg00048](https://lists.gnu.org/archive/html/bug-bash/2021-03/msg00048.html)), the error message by `"$(dirname "$(base`<kbd>Tab</kbd> happens even without `bash-completion` in Bash 5.1. While, the error message by `ls -- ${`<kbd>Tab</kbd> happens only when `bash-completion` is loaded.
2. The error message with `"$(dirname "$(base`<kbd>Tab</kbd> happens with Bash 5.1 but not with Bash 5.0 or smaller version of Bash. While, this issue was created on 2018-01-05 which predates Bash 5.1 (2020-12) and Bash 5.0 (2019-01).

Actually, the issue reported here on 2018-01 is related to my pull request #492 (which was somehow neglected by @scop). After the fix #492, this error message after `ls -- '${`<kbd>Tab</kbd> doesn't happen. On the other hand, the issue reported by @OliverLew on 2021-03 is clearly a bug of Bash so should be discussed in `bug-bash` mailing list.

**Comment by OliverLew** [comment]
@akinomyoga Thanks for the analysis and more testing with different versions of bash. Well then it's right to file the bug report to bash then :)
